Лаб. 3 6413 Волков Владислав

comandir26 opened the pull request:
**ФИО:** Волков Владислав 
**Номер группы:** 6413  
**Номер лабораторной:** 3 
**Номер варианта:** 63 
**Краткое описание предметной области:** Пункт велопроката
**Краткое описание добавленных фич:** Хранение данных было переделано с InMemory коллекций на базу данных MongoDB, интеграция EF Core с MongoDB провайдером для работы с данными через ORM. Настроен оркестратор Aspire на запуск сервера и базы данных.

kolyanch commented on CarRentalApp/CarRentalApp.API/Controllers/CarController.cs:
```diff
@@ -0,0 +1,80 @@
+    {
+        var car = mapper.Map<Car>(obj);
+        repository.Post(car);
+        return Ok();
```
Для Post-методов во всех контроллерах нужно либо использовать метод `CreatedAt`. Либо возвращать созданный объект. Иначе в данном случае имеется несоответствие метаданных по реальному возвращаемому значению (с пустым телом) и описанию метаданных в ActionResult

kolyanch commented on CarRentalApp/CarRentalApp.API/DTO/ClientDTO.cs:
```diff
@@ -0,0 +1,22 @@
+﻿namespace CarRentalApp.API.DTO;
+
+/// <summary>
+/// Класс, предоставляющий информацио об объекте Cliebt для передачи данных.
```
*информацию

kolyanch replied:
*Client

kolyanch commented on CarRentalApp/CarRentalApp.API/DTO/ClientDTO.cs:
```diff
@@ -0,0 +1,22 @@
+public class ClientDto
+{
+    /// <summary>
+    /// номер паспорта 
```
*Номер

kolyanch commented on CarRentalApp/CarRentalApp.API/DTO/RentalPlaceDto.cs:
```diff
@@ -0,0 +1,17 @@
+﻿namespace CarRentalApp.API.DTO;
+
+/// <summary>
+/// Класс, предоставляющий информацио об объекте RentalPlace для передачи данных.
```
*информацию

kolyanch commented on CarRentalApp/CarRentalApp.API/DTO/RentedCarDTO.cs:
```diff
@@ -0,0 +1,44 @@
+namespace CarRentalApp.API.DTO;
+
+/// <summary>
+/// Класс, предоставляющий информацио об объекте RentedCar для передачи данных.
```
информацию

kolyanch commented on CarRentalApp/CarRentalApp.API/DTO/RentedCarDTO.cs:
```diff
@@ -0,0 +1,44 @@
+/// </summary>
+public class RentedCarDto
+{
+    /// <summary>
+    /// выданный автомобиль 
+    /// </summary>
+    public required Car Car { set; get; }
+
+    /// <summary>
+    /// пункт проката
+    /// </summary>
+    public required RentalPlace RentalPlace { set; get; }
+
+    /// <summary>
+    /// клиент получивший машину
+    /// </summary>
+    public required Client Client { set; get; }
+
+    /// <summary>
+    /// время выдачи
+    /// </summary>
+    public required DateTime IssueTime { set; get; }
+
+    /// <summary>
+    /// срок аренды
+    /// </summary>
+    public TimeSpan? RentalPeriod { set; get; }
+
+    /// <summary>
+    /// время возврата
+    /// </summary>
+    public DateTime? ReturnTime { set; get; }
+
+    /// <summary>
+    /// пункт, в который вернули автомобиль
+    /// </summary>
+    public RentalPlace? ReturnRentalPlace { set; get; }
```
Комменты пишем с большой буквы

kolyanch commented on CarRentalApp/CarRentalApp.Domain/Client.cs:
```diff
@@ -0,0 +1,36 @@
+    /// <summary>
+    /// Номер паспорта 
+    /// </summary>
+    [Column("pusport_number")]
```
*passport

kolyanch commented on CarRentalApp/CarRentalApp.Domain/Repositories/ClientRepository.cs:
```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.EntityFrameworkCore;
```
Сделать cleanup

kolyanch reviewed (changes requested):
Исправлены не все замечания с предыдущего ревью

kolyanch commented on CarRentalApp/CarRentalApp.API/DTO/CarDTO.cs:
Почему-то все DTO-классы задублировались. Нужно удалить лишниые

comandir26 committed:
Initial commit

comandir26 committed:
Add classes

comandir26 committed:
Fix classes, add comments

comandir26 committed:
Add test data

comandir26 committed:
Add tests

comandir26 committed:
Fix tests and add comments

comandir26 committed:
Add dotnet_tests.yml for automatic testing

comandir26 committed:
Fix dotnet_tests

comandir26 committed:
Fix dotnet_tests again

comandir26 committed:
Move github directory

comandir26 committed:
Move github directory and change dotnet_tests file again x2

comandir26 committed:
Small fixes

comandir26 committed:
Corrected all comments

comandir26 committed:
Updated readme and comments

comandir26 committed:
Marked methods as static where needed

comandir26 committed:
Add IRepository interface class for inmemory repo classes, also add Bikes.Infrastructure.InMemory project for repo classes

comandir26 committed:
Implement inmemory repo classess

comandir26 committed:
Add DTO classes, add interfaces for service classes and their implementation

comandir26 committed:
Add Bikes.Api.Host project and comments

comandir26 committed:
Update tests and fix comments

comandir26 committed:
Update readme

comandir26 committed:
add primary constructors, collection expressoins
use ProducesResponseTypeautomapper, automapper
all services now returning DTO
and small fixes

comandir26 committed:
Fix api

comandir26 committed:
Fix dto's and small fixes

comandir26 committed:
Fix tests

comandir26 committed:
Validation of DTOs and fix creating objects

comandir26 committed:
some validation checks

comandir26 committed:
Fix create methods

comandir26 committed:
Add new projects. Add DbContext and repositories.

comandir26 committed:
Success build Infrastructure.MongoDb, update program.cs and extensions

comandir26 committed:
everything is working, but need to use Entity Framework Core(

comandir26 committed:
Using EfCore

comandir26 committed:
Update readme

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/Repositories/MongoBikeModelRepository.cs:
```diff
@@ -0,0 +1,92 @@
+{
+    private readonly BikesDbContext _context;
+
+    public MongoBikeModelRepository(BikesDbContext context)
```
Primary constructor

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/Repositories/MongoBikeModelRepository.cs:
```diff
@@ -0,0 +1,92 @@
+    /// <returns>List of existing objects</returns>
+    public List<BikeModel> ReadAll()
+    {
+        return _context.BikeModels.ToList();
```
```suggestion
        return [.. _context.BikeModels];
```

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/Repositories/MongoBikeRepository.cs:
```diff
@@ -0,0 +1,92 @@
+{
+    private readonly BikesDbContext _context;
+
+    public MongoBikeRepository(BikesDbContext context)
```
Аналогично

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/Repositories/MongoBikeRepository.cs:
```diff
@@ -0,0 +1,92 @@
+    /// <returns>List of existing objects</returns>
+    public List<Bike> ReadAll()
+    {
+        return _context.Bikes.ToList();
```
collection expression

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/Repositories/MongoRenterRepository.cs:
```diff
@@ -0,0 +1,91 @@
+{
+    private readonly BikesDbContext _context;
+
+    public MongoRenterRepository(BikesDbContext context)
```
Аналогично

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/Repositories/MongoRenterRepository.cs:
```diff
@@ -0,0 +1,91 @@
+    /// <returns>List of existing objects</returns>
+    public List<Renter> ReadAll()
+    {
+        return _context.Renters.ToList();
```
аналогично

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/Repositories/MongoRentRepository.cs:
```diff
@@ -0,0 +1,93 @@
+{
+    private readonly BikesDbContext _context;
+
+    public MongoRentRepository(BikesDbContext context) 
```
аналогично

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/Repositories/MongoRentRepository.cs:
```diff
@@ -0,0 +1,93 @@
+    /// <returns>List of existing objects</returns>
+    public List<Rent> ReadAll()
+    {
+        return _context.Rents.ToList();
```
аналогично

DmitryKrakhmalev commented on Bikes/Bikes.Infrastructure.MongoDb/MongoDbSeeder.cs:
```diff
@@ -0,0 +1,45 @@
+    private readonly BikesDbContext _context;
+    private readonly ILogger<MongoDbSeeder> _logger;
+
+    public MongoDbSeeder(BikesDbContext context, ILogger<MongoDbSeeder> logger)
```
аналогично

DmitryKrakhmalev commented on Bikes/Bikes.AppHost/Program.cs:
```diff
@@ -0,0 +1,9 @@
+var mongodb = builder.AddMongoDB("mongodb")
+    .WithDataVolume();
+
+var api = builder.AddProject<Projects.Bikes_Api_Host>("bikes-api")
```
```suggestion
_ = builder.AddProject<Projects.Bikes_Api_Host>("bikes-api")
```

DmitryKrakhmalev reviewed (changes requested):
Мелкие замечания которые нужно поправить.

comandir26 committed:
Small fixes
